projects/jdwp/codegen/dataclass_generator.py

```diff
@@ -0,0 +1,94 @@
+from textwrap import dedent
+
+from projects.jdwp.defs.schema import (
+    Array,
+    ArrayLength,
+    Command,
+    IntegralType,
+    PrimitiveType,
+    Struct,
+    TaggedUnion,
+)
+
+
+def map_primitive_type(primitive_type):
+    match primitive_type:
+        case PrimitiveType.STRING:
+            return "str"
+        case PrimitiveType.BOOLEAN:
+            return "bool"
+        case PrimitiveType.REFERENCE_TYPE_ID:
+            return "ReferenceTypeId"
+        case PrimitiveType.CLASS_LOADER:
+            return "ClassLoaderId"
+        case PrimitiveType.FIELD_ID:
+            return "FieldId"
+        case PrimitiveType.METHOD_ID:
+            return "MethodId"
+        case PrimitiveType.VALUE:
+            return "Value"
+        case PrimitiveType.INTERFACE_ID:
+            return "InterfaceId"
+        case PrimitiveType.CLASS_OBJECT_ID:
+            return "ClassObjectId"
+        case PrimitiveType.TAGGED_OBJECT_ID:
+            return "TaggedObjectId"
+        case PrimitiveType.THREAD_ID:
+            return "ThreadId"
+        case PrimitiveType.THREAD_GROUP_ID:
+            return "ThreadGroupId"
+        case PrimitiveType.OBJECT_ID:
+            return "ObjectId"
+        case PrimitiveType.LOCATION:
+            return "Location"
+        case _:
+            return f"Unrecognized type: {primitive_type}"
+
+
+def get_python_type_for_field(field_type):
+    if isinstance(field_type, PrimitiveType):
+        return map_primitive_type(field_type)
+    elif isinstance(field_type, IntegralType) or isinstance(field_type, ArrayLength):
+        return "int"
+    elif isinstance(field_type, Array):
+        element_type = get_python_type_for_field(field_type.element_type)
+        return f"List[{element_type}]"
+    elif isinstance(field_type, TaggedUnion):
+        pass
+    elif isinstance(field_type, Struct):
+        return "NestedStruct"
+    else:
+        raise Exception(f"Unrecognized type: {field_type}")
```

Comment:
Now that we have [`types.python_type_for`](https://github.com/facebookexperimental/ExtendedAndroidTools/blob/main/projects/jdwp/codegen/types.py#L14) could you fold new cases into it (`ArrayLength`, `Array`) ? There also are `UnionTag` and `TaggedUnion` to handle. We should also talk about handling nested structs.